### PR TITLE
build: allow passing debug CMake cache entries

### PR DESCRIPTION
**Pull request by yaelharel**
Description:
Passing "CFLAGS": "-DDEBUGBUILD" breaks the build of nghttp2 on Windows. It looks like we should be able to get the same behavior by setting "ENABLE_DEBUG": "on" in the cmake_cache entries.

Risk Level:
Low

Testing:
bazel build //source/... && bazel test //test/...
bazel build -c opt //source/... && bazel test -c opt //test/...

Docs Changes:
N/A

Release Notes:
N/A

### CODE REVIEW

**Comment by htuch** [comment]
@yaelharel can you confirm that if you run something like `http2_integration_test` at `-l trace` that you observe output from `nghttp2` library in the logs? If you could pastebin or share some example output that would give me confidence in this.

**Comment by yaelharel** [comment]
Attached are the outputs of the following runs:
```
- bazel test --test_arg="-l trace" //test/integration:http2_integration_test
- bazel test -c opt --test_arg="-l trace" //test/integration:http2_integration_test
```
They show the nghttp2 logging is working correctly, meaning we can see its logs in the output of the first run and we don't see its logs in the output of the second.
[fastbuild.log](https://github.com/envoyproxy/envoy/files/2912238/fastbuild.log) - please note, the full log was too big so we uploaded the output from one test only.
[opt.log](https://github.com/envoyproxy/envoy/files/2912222/opt.log)

File: bazel/envoy_build_system.bzl [hunk 1]
```diff
@@ -213,6 +214,11 @@ def envoy_cmake_external(
     # On Windows, we don't want to explicitly set CMAKE_BUILD_TYPE,
     # rules_foreign_cc will figure it out for us
     cache_entries_no_build_type = {key: cache_entries[key] for key in cache_entries.keys() if key != "CMAKE_BUILD_TYPE"}
+    cache_entries_no_build_type_debug = dict(cache_entries_no_build_type)
```
Comment (htuch) [inline bazel/envoy_build_system.bzl]: Suggest simplifying this by just having a function to filter out Windows entries, i.e. `filter_windows_keys` and apply it directly in the `cache_entries` dict literal below. That way, we're not tracking 4 different dictionaries, which is a bit confusing.

File: bazel/envoy_build_system.bzl [hunk 2]
```diff
@@ -213,6 +214,11 @@ def envoy_cmake_external(
...
+    cache_entries_debug = dict(cache_entries)
+    for key in debug_cache_entries.keys():
```
Comment (htuch) [inline bazel/envoy_build_system.bzl]: Nit: you should just be able to do `foo.update(bar)` for dictionary update in Python/Starlark.

**Comment by yaelharel** [comment]
@htuch Addressed feedback in the last commit.

**Pull request merged**